src/bundles/csg/functions.ts

```diff
@@ -0,0 +1,823 @@
+  union as _union,
+} from '@jscad/modeling/src/operations/booleans';
+import { extrudeLinear } from '@jscad/modeling/src/operations/extrusions';
+import {
+  align,
+  center,
+  mirror,
+  rotate as _rotate,
+  scale as _scale,
+  translate as _translate,
+} from '@jscad/modeling/src/operations/transforms';
+import { Core } from './core.js';
+import { Color, CoordinatesXYZ, Solid } from './types';
+import { clamp, hexToColor, RenderGroup, Shape } from './utilities';
+
+/* [Exports] */
+
+// [Variables - Primitive shapes]
+
+/**
+ * Primitive Shape of a cube.
+ *
+ * @category Primitive
+ */
+export const cube: Shape = shapeSetOrigin(
+  new Shape(primitives.cube({ size: 1 }))
+);
+
+/**
+ * Primitive Shape of a sphere.
+ *
+ * @category Primitive
+ */
+export const sphere: Shape = shapeSetOrigin(
+  new Shape(primitives.sphere({ radius: 0.5 }))
+);
+
+/**
+ * Primitive Shape of a cylinder.
+ *
+ * @category Primitive
+ */
+export const cylinder: Shape = shapeSetOrigin(
+  new Shape(primitives.cylinder({ radius: 0.5, height: 1 }))
+);
+
+/**
+ * Primitive Shape of a prism.
+ *
+ * @category Primitive
+ */
+export const prism: Shape = shapeSetOrigin(
+  new Shape(extrudeLinear({ height: 1 }, primitives.triangle()))
+);
+
+/**
+ * Primitive Shape of an extruded star.
+ *
+ * @category Primitive
+ */
+export const star: Shape = shapeSetOrigin(
+  new Shape(extrudeLinear({ height: 1 }, primitives.star({ outerRadius: 0.5 })))
+);
+
+//TODO
+let small = 10 ** -30;
+
+/**
+ * Primitive Shape of a square pyramid.
+ *
+ * @category Primitive
+ */
+export const pyramid: Shape = shapeSetOrigin(
+  new Shape(
+    primitives.cylinderElliptic({
+      height: 1,
+      startRadius: [0.5, 0.5],
+      endRadius: [small, small],
+      segments: 4,
+    })
+  )
+);
+
+/**
+ * Primitive Shape of a cone.
+ *
+ * @category Primitive
+ */
+export const cone: Shape = shapeSetOrigin(
+  new Shape(
+    primitives.cylinderElliptic({
+      height: 1,
+      startRadius: [0.5, 0.5],
+      endRadius: [small, small],
+    })
+  )
+);
+
+/**
+ * Primitive Shape of a torus.
+ *
+ * @category Primitive
+ */
+export const torus: Shape = shapeSetOrigin(
+  new Shape(primitives.torus({ innerRadius: 0.125, outerRadius: 0.375 }))
+);
+
+/**
+ * Primitive Shape of a rounded cube.
+ *
+ * @category Primitive
+ */
+export const rounded_cube: Shape = shapeSetOrigin(
+  new Shape(primitives.roundedCuboid({ size: [1, 1, 1] }))
+);
+
+/**
+ * Primitive Shape of a rounded cylinder.
+ *
+ * @category Primitive
+ */
+export const rounded_cylinder: Shape = shapeSetOrigin(
+  new Shape(primitives.roundedCylinder({ height: 1, radius: 0.5 }))
+);
+
+/**
+ * Primitive Shape of a geodesic sphere.
+ *
+ * @category Primitive
+ */
+export const geodesic_sphere: Shape = shapeSetOrigin(
+  new Shape(primitives.geodesicSphere({ radius: 0.5 }))
+);
+
+// [Variables - Colours]
+
+/**
+ * A hex colour code for black (#000000).
+ *
+ * @category Colour
+ */
+export const black: string = '#000000';
```

Comment:
- With the new improved support for multiple imports and for mixing modules to run them together, the repeated names would indeed be a concern I hadn't really considered
- For this, we could depend on #59 and source-academy/js-slang#618 should the user really wish to use multiple modules at the same time
- It may not be a small task to refactor things into a separate colour module as well as make the necessary updates, given that:
  - Rune colours are functions, while CSG colours are constants (which used to be hexadecimal numbers, but were changed to strings to simplify the API)
  - There is not a complete overlap between the colours the modules offer. Rune colours of the same name largely differ in the specific colours they represent. For CSG, the 16 default colours follow a pattern such that they specifically match up with one another
  - All existing materials and activities, such as slides and missions, would need to have their code and code blocks updated with the new imports. Any colour changes may also render existing images outdated

Reply:
I'm aware that imports with aliases is a supported feature, but the editor currently doesn't do syntax highlighting for it, and in general the way Source is taught in CS1101 doesn't also highlight this particular feature/issue